### PR TITLE
feat: use `account` component for brownfield env

### PR DESCRIPTION
**Pull request by nitrocode**
## what

- feat: use `account` component for brownfield env

## why

- Allow brownfield environments to use the account component without managing org, org unit, or account creation
- This will allow adding to the account outputs so account-map can ingest it without any changes

## references

- Slack conversations in sweetops https://sweetops.slack.com/archives/C031919U8A0/p1702135734967949

## tests

I tested with these toggled to `true` and `false` to ensure it worked as expected. When these are `true`, the resources are all created. When these are `false`, none of the resources are created and all the outputs are filled with existing account information with the ability to override using the yaml inputs.

```yaml
        organization_enabled: true
        organizational_units_enabled: true
        accounts_enabled: true
```

### CODE REVIEW

File: modules/account/variables.tf [hunk 1]
```diff
@@ -39,3 +39,21 @@ variable "organization_enabled" {
   description = "A boolean flag indicating whether to create an Organization or use the existing one"
   default     = true
 }
+
+variable "organizational_units_enabled" {
+  type        = bool
+  description = "A boolean flag indicating whether to create an Organizational Units"
```
[the remark this answers is not in the document]
Comment (nitrocode) [reply modules/account/variables.tf]: Thanks Andriy. I resolved it.

**Review by Nuru — CHANGES_REQUESTED** [review]
I am vary wary of making such big changes to this component, because this component is the hardest of all to test. It has to start from only a single account and create an org, OUs, and accounts, and if anything goes wrong, we cannot just delete all that and start over. 

I would prefer we support brownfield deployments with either a [static remote state configuration](https://github.com/cloudposse/terraform-yaml-stack-config/blob/40c865c14d2d707d620a493a5a279593fc8be5aa/examples/remote-state/stacks/catalog/terraform/test-component-override.yaml#L24-L30) or a parallel component that is read-only (data sources) and kept in sync with this component.

**Comment by Nuru** [comment]
I'm closing this as "wontfix" because this is such a sensitive component and so difficult to test. As I said before, I would prefer we support brownfield deployments with either a [static remote state configuration](https://github.com/cloudposse/terraform-yaml-stack-config/blob/40c865c14d2d707d620a493a5a279593fc8be5aa/examples/remote-state/stacks/catalog/terraform/test-component-override.yaml#L24-L30) or a parallel component that is read-only (data sources) and kept in sync with this component.

**Comment by nitrocode** [comment]
Thanks Jeremy for reviewing. Apologies for not responding sooner.

Options

1. Modified existing component(s) to support both brown and green
    - Allows creating new accounts and adopting old accounts
    - For what it's worth, I've used this flavor of the account-map now since proposing it and there have been no issues with maintaining existing accounts.
    - if the `account` component is so delicate/brittle, is there a plan to make it more stable by refactoring in the future? Is there plans for creating tests around components?
1. data sources would work provided it contains overrides for name and other attributes because brownfield environments often do not conform to a naming convention and need to conform in atmos. E.g. we may have an account called "Development Services" which may then need to map to "dev" in atmos. Renaming aws accounts officially can be a political problem at times to solve.
    - This would only work for migrating existing accounts and not creating new ones
1. A separate component. My concern is that the logic in this component is already quite complex to generate the appropriate outputs. If we copied and pasted that same logic into a new component, now we will need to maintain that same output logic in two components. If the logic for `account` component is updated then we will need to make the same updates to `account-brownfield` component
1. A static remote state. This would be cumbersome to maintain. There is a lot of duplicate names, account ids, etc to maintain. It might be the simplest approach. The downside is that now we cannot create new aws accounts using the component. We can only migrate existing accounts in.

Each of the options above have pros and cons my preference is to modify the existing component unless there is something im not understanding of the above options. Maybe there is another option? 

If we can test out the account flavor, even if this component is brittle, if its battle tested, then couldn't we merge the changes?

**Pull request closed without merging**